packages/salesforcedx-lightning-lsp-common/src/baseContext.ts

```diff
@@ -0,0 +1,649 @@
+      break;
+  }
+  return modulesDirs;
+};
+
+/**
+ * Holds information and utility methods for a workspace
+ */
+export abstract class BaseWorkspaceContext {
+  public type!: WorkspaceType;
+  public workspaceRoots: string[];
+
+  protected findNamespaceRootsUsingTypeCache: () => Promise<{ lwc: string[]; aura: string[] }>;
+  public initSfdxProjectConfigCache: () => SfdxProjectConfig;
+  public fileSystemProvider: FileSystemDataProvider;
+  /**
+   * @param workspaceRoots
+   * @return BaseWorkspaceContext representing the workspace with workspaceRoots
+   */
+  constructor(workspaceRoots: string[] | string, fileSystemProvider: FileSystemDataProvider) {
+    this.workspaceRoots = typeof workspaceRoots === 'string' ? [path.resolve(workspaceRoots)] : workspaceRoots;
+
+    this.findNamespaceRootsUsingTypeCache = utils.memoize(() => this.findNamespaceRootsUsingType());
+    this.initSfdxProjectConfigCache = utils.memoize(() => this.initSfdxProject());
+    this.fileSystemProvider = fileSystemProvider;
+  }
+
+  /**
+   * Initialize the workspace context asynchronously
+   */
+  public async initialize(): Promise<void> {
+    this.type = await detectWorkspaceType(this.workspaceRoots, this.fileSystemProvider);
+    if (this.type === 'SFDX') {
+      void this.initSfdxProjectConfigCache();
+    }
+  }
+
+  public async isAuraMarkup(document: TextDocument): Promise<boolean> {
+    const extension = utils.getExtension(document);
+    const isAuraExtension = AURA_EXTENSIONS.includes(extension);
+    const languageIdMatches = document.languageId === 'html' || (isAuraExtension && !document.languageId);
+    return languageIdMatches && isAuraExtension && (await this.isInsideAuraRoots(document));
+  }
+
+  public async isLWCTemplate(document: TextDocument): Promise<boolean> {
+    return (
+      document.languageId === 'html' &&
+      utils.getExtension(document) === '.html' &&
+      (await this.isInsideModulesRoots(document))
+    );
+  }
+
+  public async isInsideAuraRoots(document: TextDocument): Promise<boolean> {
+    const file = utils.toResolvedPath(document.uri);
+    for (const ws of this.workspaceRoots) {
+      if (utils.pathStartsWith(file, ws)) {
+        const isInsideAuraRoots = await this.isFileInsideAuraRoots(file);
+        return isInsideAuraRoots;
+      }
+    }
+    return false;
+  }
+
+  public async isInsideModulesRoots(document: TextDocument): Promise<boolean> {
+    const file = utils.toResolvedPath(document.uri);
+    for (const ws of this.workspaceRoots) {
+      if (utils.pathStartsWith(file, ws)) {
+        return this.isFileInsideModulesRoots(file);
+      }
+    }
+    return false;
+  }
+
+  public async isFileInsideModulesRoots(file: string): Promise<boolean> {
+    const namespaceRoots = await this.findNamespaceRootsUsingTypeCache();
+    for (const root of namespaceRoots.lwc) {
+      if (utils.pathStartsWith(file, root)) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
here and elsewhere.   
```suggestion
    return (await this.findNamespaceRootsUsingTypeCache()).some(root=>utils.pathStartsWith(file, root))
    
```

hopefully not a method eventually